analyses/collapse-rnaseq/cwl/scripts/liftover_collapse_rnaseq.py

```diff
@@ -0,0 +1,180 @@
+    """
+    for gene_sym in dup_dict:
+        means = []
+        for data in dup_dict[gene_sym]:
+            means.append(mean(data))
+        top_idx = means.index(max(means))
+        out_data = gene_sym + "\t" + "\t".join(list(map(str, dup_dict[gene_sym][top_idx]))) + '\n'
+        out_file.write(out_data)
+
+
+if __name__ == '__main__':
+    
+    parser = argparse.ArgumentParser(
+        description="Script to use a GTF and expression matrix with ENSG to liftover gene symbols."
+    )
+    parser.add_argument(
+        "-t",
+        "--table",
+        action="store",
+        dest="table",
+        help="Input expression matrix. Can be gzipped or plain tsv",
+    )
+    parser.add_argument(
+        "-o", "--output-basename", action="store", dest="out", help="output basename of liftover and collapse if flag given"
+    )
+    parser.add_argument(
+        "-i", "--input-gtf", action="store", dest="gtf", help="GTF file to use as liftover reference. Can be gzipped or plain tsv"
+    )
+    parser.add_argument(
+        "-g", "--gene-id", action="store", dest="gene_id", help="NAME of gene ID (ENSG values) column"
+    )
+    parser.add_argument(
+        "-n", "--gene-name", action="store", dest="gene_name", help="NAME of gene name column, if present to keep if not found"
+    )
+    parser.add_argument(
+        "-s", "--skip", action="store", dest="skip", type=int, help="Number of lines to skip if needed"
+    )
+    parser.add_argument(
+        "-c", "--collapse", action="store_true", dest="collapse", help="If set, will collapse on gene_symbol"
+    )
+
+    args = parser.parse_args()
+
+    gtf_dict = {}
+    # hardcoded for now, could make opts
+    tx_type = 2
+    tx_desc = 8
+    # set up pattern matcher ahead of time for efficiency
+    id_find = re.compile(r"gene_id \"(ENSG\d+)\.\d+\";")
+    name_find = re.compile(r".*gene_name \"(\S+)\";")
+    sys.stderr.write('Indexing gtf\n')
+    with (gzip.open if args.gtf.endswith("gz") else open)(args.gtf, "rt", encoding="utf-8") as gtf:
+        for line in gtf:
+            if not line.startswith('##'):
+                update_gtf_dict(gtf_dict, line)
+    # track sym occurrences if collapse flag given for later processing
+    sym_ct = {}
+    with (gzip.open if args.table.endswith("gz") else open)(args.table, "rt", encoding="utf-8") as table:
+        if args.skip:
+            sys.stderr.write('Skipping ' + str(args.skip) + ' lines\n')
+            for i in range(args.skip):
+                skip = next(table)
+        head = next(table)
+        header = head.rstrip('\n').split('\t')
+        # get positions of current gene id and name
+        g_idx = header.index(args.gene_id)
+        n_idx = header.index(args.gene_name)
+        # make assumption that data will start after ID and name
+        d_idx = g_idx + 1
+        if n_idx > g_idx:
+            d_idx = n_idx + 1
+        sys.stderr.write('Processing table\n')
+        sys.stderr.flush()
```

Comment:
Wouldn't it just make more sense to use `print("Processing table", file=sys.stderr)`. Then you don't have to worry about having `\n` or `flush()`.